kani-compiler/src/codegen_aeneas_llbc/mir_to_ullbc/mod.rs

```diff
@@ -409,54 +751,118 @@ impl<'a, 'tcx> Context<'a, 'tcx> {
         instance_internal.def.requires_caller_location(self.tcx())
     }
 
-    fn translate_ty(&self, ty: Ty) -> CharonTy {
+    fn translate_generic_args(&mut self, ga: GenericArgs) -> CharonGenericArgs {
```

Comment:
There's some redundancy between this function and `generic_params_from_adtdef` above. Is it possible to somehow combine them to avoid code duplication?

Reply:
I think they are quite different and pretty hard to combine.